### PR TITLE
test: Create user journey for request rejection

### LINKED ISSUE #910: Create a user journey test from sending a mentorship request and the request is rejected.
## Description
I need to test a specific user path,
so that I can make sure a specific path can be executed usefully in a deterministic way.

This test should be located in a specific folder for separation concerns, it could be called tests/user_journey.
Next, I'll define a very broad scenario, but simple enough for you to start.

## Scenario: User A (mentor) and User B (mentee)

  1. User B (mentee) sends a request to User A (mentor)
  2. User A (mentor) rejects request

## Definition of Done
- [ ] Create 1 user journey test that follows the scenario described above
- [ ] Document the test result on Test Quality (@anitab-org/qa-team can help)

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by sankha555**
### Description

This patch creates a user journey test depicting the process of sending a new mentorship request by a mentee to a mentor. The request is received by the mentor and it is rejected by them. The mentee then checks and discovers that the request has been rejected. At each stage mentioned above, the corresponding consistency checks are made and relevant messages are also printed to the console. The code has been commented for better readability and understanding.

Fixes #910

### Type of Change:

- Code

### How Has This Been Tested?

Tested the user journey script added in this patch on my local system and it returned OK status. 

image

Also ran unittests and they returned OK status.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Review by rpattath — DISMISSED** [review]
LGTM ! Thank you @sankha555

File: tests/user_journey/test_rejection_path.py
Comment (isabelcosta) [inline tests/user_journey/test_rejection_path.py]: Can you just remove the `.value` from the HTTPStatus enums please. It works either way in this tests situation so it should be fine without it ;) I can approve after.
```suggestion
        self.assertEqual(HTTPStatus.OK, reject_response.status_code)
```

File: tests/user_journey/test_rejection_path.py
Comment (isabelcosta) [inline tests/user_journey/test_rejection_path.py]:
```suggestion
            HTTPStatus.OK, mentee_current_rejected_relation.status_code
```

File: tests/user_journey/test_rejection_path.py
Comment (isabelcosta) [inline tests/user_journey/test_rejection_path.py]:
```suggestion
        self.assertEqual(HTTPStatus.CREATED, send_request_response.status_code)
```

**Review by isabelcosta — DISMISSED** [review]
Great work @sankha555 !
I just did minor changes, I should have been more clear about, in which the `.value` is only needed for that special case of documenting responses on Swagger UI.

Thank you for this contribution!

**Comment by isabelcosta** [comment]
@sankha555 can you just run black against these changes please? I have removed the `.value` which makes some lines not compliant with black formatting. After that, I am happy to merge this :)

**Comment by sankha555** [comment]
@isabelcosta extremely sorry for the late response, I had been busy with some university assignments for the past few days and couldn't give time here. I have made the changes, kindly take a look. Thanks! :)

File: tests/user_journey/test_rejection_path.py
Comment (isabelcosta) [inline tests/user_journey/test_rejection_path.py]: is this doc string updated? @sankha555

Comment (sankha555) [reply tests/user_journey/test_rejection_path.py]: Sorry for that @isabelcosta , I have changed the comment now. Please take a look. Thanks!

**Review by isabelcosta — APPROVED** [review]
Great work @sankha555 ! Thank you again :D

**Review by vj-codes — APPROVED** [review]
LGMT 🎉

**Comment by isabelcosta** [comment]
@vj-codes Will merge since this is a test and naturally is being tested through "Run tests" GitHub workflow. Thank you for reviewing 🙌🏾

**Pull request merged**
